### PR TITLE
Allow ReduceLROnPlateau to track val_loss when validation set is available

### PR DESCRIPTION
**Pull request by gorold**
*Description of changes:*
Allow ReduceLROnPlateau to track val_loss when validation set is available. This makes the torch training process to be more in line with the current mxnet training process.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

**Please tag this pr with at least one of these labels to make our release process faster:** BREAKING, new feature, bug fix, other change, dev setup

### CODE REVIEW

**Review by shubhamkapoor — APPROVED** [review]
Looks good to me! 🚀

**Comment by lostella** [comment]
I think we can backport this to 0.12

**Review by shchur — APPROVED** [review]
Nice, thank you!
